packages/calcite-components/src/utils/form.tsx

```diff
@@ -519,6 +550,16 @@ function defaultSyncHiddenFormInput(
   }
 
   component.syncHiddenFormInput?.(input);
+
+  const validationComponent = getValidationComponent(component.el as HTMLCalciteInputElement);
+
+  if (validationComponent && "validity" in validationComponent) {
+    // mutate the component's validity object to prevent a rerender
+    // https://stenciljs.com/docs/properties#mutable-arrays-and-objects
+    for (const key in { ...input?.validity }) {
+      validationComponent.validity[key] = input?.validity[key];
```

Comment:
Given the keys are coming from `input`, do you need to check `input`'s existence here?